### PR TITLE
integrate crypto concepts into jrt

### PR DESCRIPTION
**Pull request by havardpe**
@bjorncs @arnej27959 please review
@vekterli FYI

### CODE REVIEW

**Review by bjorncs — APPROVED** [review]

File: jrt/src/com/yahoo/jrt/CryptoEngine.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+/**
+ * Component responsible for wrapping low-level sockets into
+ * appropriate CryptoSocket instances. This is the top-level interface
+ * used by code wanting to perform network io with appropriate
+ * encryption.
+ **/
+public interface CryptoEngine {
+    public CryptoSocket createCryptoSocket(SocketChannel channel, boolean isServer);
```
Comment (bjorncs) [inline jrt/src/com/yahoo/jrt/CryptoEngine.java]: The signature should allow implementations to throw `IOException` if creating the socket fails.

Comment (havardpe) [reply jrt/src/com/yahoo/jrt/CryptoEngine.java]: the socket is already created, and passed in as 'channel'

**Review by arnej27959 — APPROVED** [review]
looks good to me

File: jrt/src/com/yahoo/jrt/CryptoEngine.java (hunk 1 above)
Comment (havardpe) [reply jrt/src/com/yahoo/jrt/CryptoEngine.java]: ok, you meant the 'CryptoSocket' itself... yes, we can probably do that.

Comment (havardpe) [reply jrt/src/com/yahoo/jrt/CryptoEngine.java]: Please note that the constructor may be called from another thread than the other methods

**Pull request merged**